pinot-core/src/main/java/org/apache/pinot/core/operator/filter/FilterOperatorUtils.java

```diff
@@ -28,9 +28,207 @@
 
 
 public class FilterOperatorUtils {
+
+  private static volatile Implementation _instance = new Implementation();
+
   private FilterOperatorUtils() {
   }
 
+  public static void setImplementation(Implementation newImplementation) {
+    _instance = newImplementation;
+  }
+
+  public static class Implementation {
```

Comment:
Let's introduce an interface, and make this the default implementation

Reply:
I would love to do that, but... how would we consume the interface from other classes? Right now we call the a static method, so we don't need any reference. If we use an interface here, we would need to supply the instance to all callers (like `FilterPlanNode`).

Do you have an idea on how to do that without adding a new argument in the constructor? As said in other PRs, this is not a new problem. We may need to start thinking about introducing a dependency injection system like Guice in Pinot, as it would be less aggressive in the code than adding so many changes on the constructors.

Reply:
To minimize the change, we can still keep the `FilterOperatorUtils` and all the static util function, but also make an interface (basically add an interface for the methods, and make this class the default implementation for the interface). I'd imagine you want to plug in some other implementation via the `setImplementation()`, instead of extending this class, you may implement the interface